shortcut.go

```diff
@@ -46,20 +48,23 @@ func New(ipv4Subnets []string, ipv6Subnets []string) Shortcut {
 	return &shortcut{
 		v4list: newSortList(ipv4Subnets),
 		v6list: newSortList(ipv6Subnets),
+		// Prefers the system resolver by default, hopefully can use OS DNS cache.
+		resolver: net.DefaultResolver,
```

Comment:
The type `Resolver` has a `PreferGo` setting that can be used to force a Go implementation of the DNS lookup (which does not cache). This is `false` by default and on Windows at least it's using syscalls to the system's resolver, so we should be fine on caching.

Reply:
Yeah verified on macOS that it uses cache too.